tools/quickstart/quickstart/utils.py

```diff
@@ -0,0 +1,143 @@
+import re
+import sys
+from pathlib import Path
+from typing import Tuple
+
+import click
+from eth_account import Account
+
+from quickstart.constants import (
+    ADDRESS_FILE_PATH,
+    BRIDGE_CONFIG_FILE_EXTERNAL,
+    KEY_DIR,
+    NETSTATS_ENV_FILE_PATH,
+    PASSWORD_FILE_PATH,
+)
+
+
+def ensure_clean_setup():
+    if os.path.isdir(KEY_DIR):
+        raise click.ClickException(
+            "The directory holding the keys already exists.\n"
+            "This should not occur during normal operation."
+        )
+
+    if os.path.isfile(PASSWORD_FILE_PATH):
+        raise click.ClickException(
+            "The password file already exists.\n"
+            "This should not occur during normal operation."
+        )
+
+
+def is_validator_account_prepared() -> bool:
+    return os.path.isfile(ADDRESS_FILE_PATH)
+
+
+def is_netstats_prepared() -> bool:
+    return (
+        os.path.isfile(NETSTATS_ENV_FILE_PATH)
+        # Ignore touched only files (docker-compose workaround)
+        and os.stat(NETSTATS_ENV_FILE_PATH).st_size != 0
+    )
+
+
+def is_bridge_prepared() -> bool:
+    return (
+        os.path.isfile(BRIDGE_CONFIG_FILE_EXTERNAL)
+        # Ignore touched only files (docker-compose workaround)
+        and os.stat(BRIDGE_CONFIG_FILE_EXTERNAL).st_size != 0
+    )
+
+
+class TrustlinesFiles:
+    def __init__(self, password_file, address_file, keystore_file):
+        self.password_file = password_file
+        self.address_file = address_file
+        self.keystore_file = keystore_file
+
+    def store(self, account, password):
+        # Instead of just copying the file, we encrypt it again since the
+        # file itself may use scrypt as key derivative function which
+        # parity can't handle
+        json_account = account.encrypt(password, kdf="pbkdf2")
+
+        os.makedirs(os.path.dirname(os.path.abspath(self.keystore_file)), exist_ok=True)
+        with open(self.keystore_file, "w") as f:
+            f.write(json.dumps(json_account))
+
+        with open(self.password_file, "w") as f:
+            f.write(password)
+
+        with open(self.address_file, "w") as f:
+            f.write(account.address)
+
+
+def is_wrong_password_error(err):
+    return type(err) is ValueError and str(err) == "MAC mismatch"
+
+
+def get_keystore_path() -> str:
+    click.echo(
+        "Please enter the path to the keystore file to import.\nIf you started the process via the quickstart scipt, "
+        "please consider that you are running within a Docker virtual file system. "
+        "You can access the current working directory via '/data'. "
+        "(e.g. './my-dir/keystore.json' becomes '/data/my-dir/keystore.json')"
+    )
+
+    while True:
+        raw_path = click.prompt("Path to keystore", default="./account-key.json")
+        absolute_path = Path(raw_path).expanduser().absolute()
+
+        if os.path.isfile(absolute_path) and os.access(absolute_path, os.R_OK):
+            return absolute_path.as_posix()
```

Comment:
what does `as_posix` do here?

Reply:
Transform the path object as String. I could change the return type to `Path` and adjust the `import_keystore_file()` for it. No strong feelings here.

Reply:
if you want to convert it to a string, using `str(absolute_path)` would be clearer

Reply:
:heavy_check_mark: